### ISSUE TITLE
Test connection Oracle DB ready - OK greyed out !

### ISSUE THREAD

**Issue by Fxztam** [qgis/QGIS#39116]
Hi,

I try to define a new Oracle Remote connection in QGis 3.14.16 and the **Test check works fine**.

Then I want to save this connection, but the <OK> button is grayed out !

I cannot set the server name - that does not work - it's a remote Cloud connection.

What is wrong ??

Thanks Friedhold

**Comment by gioman** [comment]
@Fxztam duplicate of https://github.com/qgis/QGIS/issues/38979 ?

**Comment by troopa81** [comment]
> @Fxztam duplicate of #38979 ?

Yes, it is. I close the issue. @Fxztam please re-open it if you think otherwise

**Comment by adepoch** [comment]
Hi. Was this ever fixed in versions after 3.16.4-Hannover? QGIS is currently unusable in my organisation due to this bug as we use Oracle as our main spatial data store. I can concur that the OK button in the Oracle connection dialog is not active until **both** a Database and Host are given. However providing a Host causes the Test Connect to fail, but it succeeds with the Host left blank. So we can't actually make connections to Oracle from QGIS. I'm not sure in this case if the bug is that proving a Host causes failure in the connection, or whether the bug is that the OK button needs both fields before it is active. Either way there's a bug.

**Comment by gioman** [comment]
> QGIS is currently unusable in my organisation due to this bug

@adepoch file a new issue (check if is not existing already) https://github.com/qgis/QGIS/issues?q=is%3Aopen+is%3Aissue+label%3A%22Oracle+data+provider%22 and since you are an organization you should consider supporting the work necessary to fix this issue.

**Comment by troopa81** [comment]
> I can concur that the OK button in the Oracle connection dialog is not active until both a Database and Host are given

I have just tested with last 3.16.9, OK button is active if
- You set only a database name  --> t's considered as a service name described in ora file
- You set a dabatase **AND** an host **AND** a port --> It's a direct connection

What type of configuration do you try to set up ? direct connection or with a service file?

I'm pretty sure this behavior was existing  in 3.16.4, but could you test it with 3.16.16 ? or new LTR 3.22.4 ?

**Comment by adepoch** [comment]
> > I can concur that the OK button in the Oracle connection dialog is not active until both a Database and Host are given
> 
> I have just tested with last 3.16.9, OK button is active if
> 
> * You set only a database name  --> t's considered as a service name described in ora file
> * You set a dabatase **AND** an host **AND** a port --> It's a direct connection
> 
> What type of configuration do you try to set up ? direct connection or with a service file?
> 
> I'm pretty sure this behavior was existing in 3.16.4, but could you test it with 3.16.16 ? or new LTR 3.22.4 ?

Thanks for the help @troopa81 . It turns out that it isn't exactly a bug, but I was doing it wrong. However I was doing it wrong because there's not much information on the dialog to indicate the level of information I should supply. If I keep a bare minimum DB name, and credentials, and DON'T put in port or server name then the OK button is activated. As the port field starts with a port number in it the assumption was that it was needed, which in turn made me think the server name was needed. So my bad in some ways, but if you're interested in usability I think the dialog could be self documented a little better. But thanks. I've updated our internal user doco.

@gioman I am not an organisation, I'm just a minion who works for one (programmer employed as a spatial data manager). If I controlled the purse strings I would love to support the project, but I'm sorry I don't. However I am proud that we have contributed to OGC before if that makes you feel better :-)
